test/api_stability/README.md

```diff
@@ -0,0 +1,149 @@
+  * But use the test cases from the LSR.
+5. Build all tests.
+6. Run all tests.
+7. If any of the above steps fail the complete build will fail.
+
+We can apply this workflow to every kind of test (e.g. `test/unit`, `test/snippet`, `test/benchmarks`) that we have.
+Currently, we only test `test/unit` and `test/snippet` as both of them test most of our public API.
+
+### When to create patches and what is the naming scheme?
+
+Patches should only be created if API changed in a way that we can't give an "automatic" upgrade path by e.g. defining a
+typedef, a compatible function that perfectly forwards to the new API or some other means that provide the old API which
+then uses the new API. Please first try your hardest to provide "compatible" API, as this is the easiest for the user.
+
+There are two categories of API changes that are reflected in the patches (prefixes in the commit message):
+
+* `[API]` are patches that changed somehow the public API (regardless whether this is `stable` or `experimental` API),
+  like the behaviour, the return type or the call convention. These patches should always have a CHANGELOG.md entry
+  explaining what the change did. There are these sub kinds of `[API]` changes:
+  * `[API] [BREAKAGE]` an API that existed and has a completely different meaning now, such a change MUST have a real
+    good reason and should be avoided or documented extremely well.
+  * `[API] [FIX]` the change was necessary, as it fixed a defect. Some examples:
+    * a function does not throw any more / will throw now
+    * format of output (e.g. std::cout) changed
+    * CTAD changed (e.g. `some_class{0}` will now be of type `some_class<int>` instead of `some_class<double>`)
+    * The resulting return type changed / order within template parameters changed
+  * `[API] [WRONG-USAGE]` an API was misused in the test or tested some invalid scenarios
+* `[NOAPI]` are patches that change internal (e.g. `seqan3::detail` or `private` member) or non-public API (e.g. marked
+  as `\noapi`). These changes don't need any upgrade-path.
+  * `[NOAPI] [INCLUDE]` means that some header file was moved, or an unnecessary include was removed in some header,
+    which caused a test to explicitly include that header. As there are tools like [Include What You
+    Use](https://github.com/include-what-you-use/include-what-you-use), we don't consider these changes as API stability
+    issues. We still will try to add an upgrade path if possible (like when we split `/include/seqan3/core` and
+    `/include/seqan3/utility`) and add CHANGELOG.md entries for every header change.
+  * `[NOAPI] [DETAIL]` means that some `seqan3::detail` code changed, like a rename, or template parameter change.
+  * `[NOAPI] [DEPRECATED]` means that some deprecated code was removed.
+  * `[NOAPI] [BREAKAGE]` means that the same API has a considerable different behaviour. E.g. before it accepted some
+    input, and now it would seg-fault on the same input.
```

Comment:
```suggestion
* `[API]` are patches that changed the public API in some way (regardless of whether this is `stable` or `experimental`
  API), like the behaviour, the return type or the call convention. These patches should always have a `CHANGELOG.md` 
  entry explaining what the change did. `[API]` changes are further divided into these subcategories:
  * `[API] [BREAKAGE]` An API that existed and has a completely different meaning now, such a change MUST have a really
    good reason and should be avoided or documented extremely well.
  * `[API] [FIX]` The change was necessary, as it fixed a defect. Some examples:
    * A function does not throw anymore / will throw now.
    * Format of output (e.g., `std::cout`) changed.
    * CTAD changed (e.g., `some_class{0}` will now be of type `some_class<int>` instead of `some_class<double>`).
    * The return type changed / order within template parameters changed.
  * `[API] [WRONG-USAGE]` An API was misused in the test or tested some invalid scenarios.
* `[NOAPI]` are patches that change internal (e.g., `seqan3::detail` or `private` member) or non-public API (e.g., 
  marked `\noapi`). These changes do not need an upgrade path. We differentiate between the following subcategories:
  * `[NOAPI] [INCLUDE]` Some header file was moved, or an unnecessary include was removed in some header,
    which caused a test to explicitly include that header. As there are tools like [Include What You
    Use](https://github.com/include-what-you-use/include-what-you-use), we do not consider these changes as API 
    stability issues. We will still try to add an upgrade path if possible (like when we split `/include/seqan3/core` 
    and `/include/seqan3/utility`) and add `CHANGELOG.md` entries for every header change.
  * `[NOAPI] [DETAIL]` Some `seqan3::detail` code changed, like a rename, or template parameter change.
  * `[NOAPI] [DEPRECATED]` Some deprecated code was removed.
  * `[NOAPI] [BREAKAGE]` The same API has a considerable different behaviour. E.g., it accepted some
    input before, but now it would segfault on the same input.
```